runners/spark/spark_runner.gradle

```diff
@@ -181,7 +181,13 @@ dependencies {
   runtimeOnly library.java.jackson_module_scala_2_12
   // Force paranamer 2.8 to avoid issues when using Scala 2.12
   runtimeOnly "com.thoughtworks.paranamer:paranamer:2.8"
-  provided library.java.hadoop_common
+  if (project.property("spark_version").startsWith("3.2")) {
```

Comment:
Spark 2.4 was dropped, it's fine to remove the if clause here and just keep hadoop-client-api.